### ISSUE TITLE
[Bug]: Indexing game files for 2 hour, and still get nothing.

### ISSUE THREAD

**Issue by moiian** [Kizari/Flagrum#165]
### Operating System

Windows 11 64-Bit

### Game Version

Latest Steam Release

### Version

1.5.14

### Description of the bug

When i change profile to Forspoken. Indexing game files in several minutes and everything worked fine. 

However, when i changed it to FFXV, I've been stuck on 'Indexing game files' for many hours.

I tried rebuilding the game file index, which caused the software to crash. I also tried reinstalling the software, but it didn't help. I have a high-performance computer and an SSD.

FFXV is the steam version. But i copy it from another PC. Not sure if it is matter.

### Steps to reproduce the bug

1.Change profile to FFXV, indexing files for 2 hour
2.Click the rebuilding the game file index
3.Crash
4.Delete and reinstall Flagrum
5.Still endless indexing
6.Change profile to Forspoken, game and mod work well

### Code of Conduct

- [X] I confirm that I have ensured this bug hasn't already been reported in another issue on this repository.
- [X] I confirm that this bug report is pertaining to Flagrum, not an individual mod, the game itself, or an issue with my system or files.
- [X] I confirm that I am using a legitimate copy of the game.
- [X] I confirm that I installed Flagrum with a clean installation of the game, and have not used any tools other than Flagrum to mod my game files.
- [X] If my issue relates to the game itself crashing or exhibiting unexpected behaviour, I confirm that I have tried running a clean unmodded installation of the game to see if this resolves my issue before filing a bug report.

**Comment by moiian** [comment]
`System.AggregateException
Message: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (Unable to read beyond the end of the stream.))))))
Stack Trace: 
System.AggregateException
Message: One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (Unable to read beyond the end of the stream.)))))
Stack Trace:    at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 219
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.RegenerateFileIndex() in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 37
   at Flagrum.Web.Services.AppStateService.<LoadNodes>b__57_0() in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Services\AppStateService.cs:line 110
System.AggregateException
Message: One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (Unable to read beyond the end of the stream.))))
Stack Trace:    at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 219
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
System.AggregateException
Message: One or more errors occurred. (One or more errors occurred. (One or more errors occurred. (Unable to read beyond the end of the stream.)))
Stack Trace:    at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 219
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
System.AggregateException
Message: One or more errors occurred. (One or more errors occurred. (Unable to read beyond the end of the stream.))
Stack Trace:    at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 219
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
System.AggregateException
Message: One or more errors occurred. (Unable to read beyond the end of the stream.)
Stack Trace:    at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 219
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
System.IO.EndOfStreamException
Message: Unable to read beyond the end of the stream.
Stack Trace:    at System.IO.BinaryReader.ReadByte()
   at Flagrum.Core.Utilities.Extensions.StreamExtensions.ReadNullTerminatedString(BinaryReader reader, Encoding encoding) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Core\Utilities\Extensions\StreamExtensions.cs:line 53
   at Flagrum.Core.Archive.EbonyArchive.ReadFileHeaders() in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Core\Archive\EbonyArchiveReader.cs:line 58
   at Flagrum.Core.Archive.EbonyArchive..ctor(String archivePath) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Core\Archive\EbonyArchive.cs:line 47
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexDirectory(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 235
   at Flagrum.Web.Features.AssetExplorer.Indexing.FileIndexer.IndexRecursively(String directory) in D:\a\Flagrum-Dev\Flagrum-Dev\Flagrum.Web\Features\AssetExplorer\Indexing\FileIndexer.cs:line 218
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
`

**Comment by moiian** [comment]
After verify the game file on steam, Flagrum work well.
